File: Java/src/main/java/QuestionBank/Solution501_600/Solution594.java
package QuestionBank.Solution501_600;

import java.util.HashMap;
import java.util.Map;

/**
 * @author windows
 */
public class Solution594 {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        int max = 0;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (map.containsKey(entry.getKey() - 1)) {
                max = Math.max(max, map.get(entry.getKey() - 1) + entry.getValue());
            }else if (map.containsKey(entry.getKey() + 1)){
                max = Math.max(max, map.get(entry.getKey() + 1) + entry.getValue());
            }
        }
        return max;
    }


    public static void main(String[] args) {
        Solution594 solution594 = new Solution594();
    }
}

File: C++/src/Solution501_600/Solution594.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution594.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_map>

int Solution594::findLHS(vector<int> &nums) {
    int n = nums.size();
    unordered_map<int, int> map;
    int max = 0;
    for (int i = 0; i < n; ++i) {
        map[nums[i]]++;
    }
    for (auto& [key, value] : map) {
        if (map.count(key - 1)){
            if (map[key - 1] + map[key] > max){
                max = map[key - 1] + map[key];
            }
        }
    }
    return max;
}
